app/models/user/permissions.rb

```diff
@@ -39,27 +40,52 @@ def admin_sections=(sections)
     keep_permissions = existing_permissions.values_at(*keep)
     new_permissions = create.map { |section| member_permissions.build(admin_section: section) }
 
-    self.member_permissions = keep_permissions + new_permissions
+    all_permissions = keep_permissions + new_permissions
+    # keep service permissions as they were
+    all_permissions << services_member_permission if services_member_permission
+
+    self.member_permissions = all_permissions
   ensure
     @_admin_sections = nil
   end
 
-  def member_permission_service_ids=(service_ids)
-    services_section = [:services]
+  # returns all permissions (:portal, :finance, :settings, :partners, :monitoring, :plans) for admins
+  # and the allowed ones for member users
+  def member_permission_ids
+    admin? ? AdminSection.permissions : admin_sections - [:services]
+  end
 
+  def member_permission_ids=(roles)
+    self.admin_sections = Array(roles).reject(&:blank?)
+  end
+
+  def member_permission_service_ids=(service_ids)
     if service_ids.present?
-      self.admin_sections = admin_sections | services_section
-      services_member_permission.service_ids = service_ids
-    else
-      self.admin_sections = admin_sections - services_section
+      if services_member_permission
```

Comment:
As @macejmic said, this method is complicated and needs refactoring, but I know that you guys are already on it 👍